sdks/wasm/Mono.WebAssembly.DebuggerProxy/MonoProxy.cs

```diff
@@ -98,58 +108,91 @@ enum StepKind {
 		Over
 	}
 
+	internal class ExecutionContext {
+		public bool RuntimeReady { get; set; }
+		public int Id { get; set; }
+		public object AuxData { get; set; }
+
+		public List<Frame> CallStack { get; set; }
+		public int LocalBreakpointId { get; set; }
+
+		internal DebugStore store;
+		public TaskCompletionSource<DebugStore> Source { get; } = new TaskCompletionSource<DebugStore> ();
+
+		public DebugStore Store {
+			get {
+				if (store == null || !Source.Task.IsCompleted)
```

Comment:
1. if the task is not complete, then this being called is likely a bug, so maybe we should throw an exception?
2. Callers never seem to check `context.Store==null`. If the assumption is that this will only be called when it has been successfully loaded, then we should throw an exception here. Else, we need to check this at the call sites.

Reply:
Throwing in that case is intensional testing for ordering constraints

Reply:
maybe rename `store` to `_store`, just to ensure that we don't accidentally use the field when the the property should have been used?

Reply:
not "ensure", but make it more apparent.